pysat/tests/test_instruments.py

```diff
@@ -103,278 +100,110 @@ def init_func_external(self):
-                f = partial(self.check_load, inst)
-                f.description = ' '.join(('Checking load routine',
-                                          'functionality for module with',
-                                          'clean level "none": ',
-                                          inst.platform, inst.name, inst.tag,
-                                          inst.sat_id))
-                yield (f,)
-
-                inst.clean_level = 'dirty'
-                f = partial(self.check_load, inst, fuzzy=True)
-                f.description = ' '.join(('Checking load routine',
-                                          'functionality for module with',
-                                          'clean level "dirty": ',
-                                          inst.platform, inst.name, inst.tag,
-                                          inst.sat_id))
-                yield (f,)
-
-                inst.clean_level = 'dusty'
-                f = partial(self.check_load, inst, fuzzy=True)
-                f.description = ' '.join(('Checking load routine',
-                                          'functionality for module with',
-                                          'clean level "dusty": ',
-                                          inst.platform, inst.name, inst.tag,
-                                          inst.sat_id))
-                yield (f,)
-
-                inst.clean_level = 'clean'
-                f = partial(self.check_load, inst, fuzzy=True)
-                f.description = ' '.join(('Checking load routine',
-                                          'functionality for module with',
-                                          'clean level "clean": ',
-                                          inst.platform, inst.name, inst.tag,
-                                          inst.sat_id))
-                yield (f,)
-
+        # check for username
+        inst_name = '_'.join((inst.platform, inst.name))
+        dl_dict = user_download_dict[inst_name] if inst_name in \
+            user_download_dict.keys() else {}
+        inst.download(start, start, **dl_dict)
+        assert len(inst.files.files) > 0
+
+    @pytest.mark.second
+    @pytest.mark.parametrize("inst", instruments['list'])
+    @pytest.mark.parametrize("clean_level", ['none', 'dirty', 'dusty',
+                                             'clean'])
+    def test_load(self, inst, clean_level):
+        # make sure download was successful
+        print(' '.join(('\nChecking load routine functionality for module',
+                        inst.platform, inst.name, inst.tag, inst.sat_id)))
+        if len(inst.files.files) > 0:
+            inst.clean_level = clean_level
+            target = 'Fake Data to be cleared'
+            inst.data = [target]
+            start = inst._test_dates[inst.sat_id][inst.tag]
+            inst.load(date=start)
+            # Make sure fake data is cleared
+            assert target not in inst.data
```

Comment:
Using `not in` here since `!=` does not work with `xarray` instruments, it will return an `xarray.Dataset` rather than a boolean.  This throws an `AssertionError`, which in the previous code was screened out by the bare `except` statement.